tools/testsys/src/vmware_k8s.rs

```diff
@@ -0,0 +1,241 @@
+use crate::aws_resources::migration_crd;
+use crate::crds::{
+    BottlerocketInput, ClusterInput, CrdCreator, CrdInput, CreateCrdOutput, MigrationInput,
+    TestInput,
+};
+use crate::error::{self, Result};
+use crate::sonobuoy::sonobuoy_crd;
+use bottlerocket_types::agent_config::{
+    CreationPolicy, K8sVersion, VSphereK8sClusterConfig, VSphereK8sClusterInfo, VSphereVmConfig,
+};
+use maplit::btreemap;
+use model::{Crd, DestructionPolicy};
+use pubsys_config::vmware::Datacenter;
+use snafu::OptionExt;
+use std::collections::BTreeMap;
+use std::str::FromStr;
+
+/// A `CrdCreator` responsible for creating crd related to `aws-k8s` variants.
+pub(crate) struct VmwareK8sCreator {
+    pub(crate) region: String,
+    pub(crate) datacenter: Datacenter,
+    pub(crate) ova_name: String,
+    pub(crate) encoded_mgmt_cluster_kubeconfig: String,
+}
+
+#[async_trait::async_trait]
+impl CrdCreator for VmwareK8sCreator {
+    /// Use the `BUILDSYS_OVA` name for the image id.
+    fn image_id(&self, _: &CrdInput) -> Result<String> {
+        Ok(self.ova_name.to_string())
+    }
```

Comment:
I understand why this must be this way but I wonder if we can generalize `image_id` a bit more so it can encompass OVA names as well.

Reply:
I'm not sure what you are asking for here.

Reply:
`image_id` as the getter interface might not be accurate for returning OVA file names. When I see "Image ID", it makes me think of cloud images and not OVA files that which acts more like a package (inside is a `vmdk` file which is the actual image). In a weird way I guess "ID" can also mean name of a file and since this is an implementation detail that is not exposed to users, we can revisit this if it ever causes any confusion.

Reply:
Thanks for clarifying.